### PR TITLE
mattdrayer/xblock-i18n: Add documentation about XBlock I18N/L10N support

### PR DESCRIPTION
**Pull request by mattdrayer**
@nedbat @lamagnifica -- First draft of XBlock I18N/L10N documentation -- happy to add/edit, just let me know what you would like to see.

### CODE REVIEW

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 1]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
   * ``video``
   * ``other``
 
+*****************************
+Internationalization Support
+*****************************
+
+The LMS is currently capable of supporting internationalization (I18N) and localization (L10N) of static XBlock chrome
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I've only seen i18n and l10n as all lowercase... Quick google search shows it in lowercase on W3C and wikipedia, so I suggest we use that standard.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 2]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+and control strings.  Translation of user-generated content stored as XBlock state is not currently supported.
+
+* In order for the LMS to present XBlock language translations, you must include your translated strings in the
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I suggest rewording "In order for the LMS to present XBlock language translations..." to "To present XBlock language translations in the LMS..."
(the goal is to put the task that the reader is trying to accomplish first)

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 3]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+  GNU Gettext Portable Object file format.
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Maybe lead in to the list with something like, For more information about this file format, see the following references.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 4]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+
+  * https://en.wikipedia.org/wiki/Gettext
+  * https://www.drupal.org/node/1814954
+
+* Create a folder within your XBlock project named "translations":
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Avoid using a list if you only have a single item to present.
Wait.
Is the URL that follows an example, or do you want them to actually follow the link? When I follow it, I get a 404, so I'd suggest marking it up as a code sample or enclosing it in double `` characters. (The rendering process will automatically turn it into a live link if you don't mark it up in one of these ways)

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 5]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+
+  * http://github.com/my_org/my_xblock/my_xblock/translations
+
+* Create a set of subfolders (locales) and PO files (domains) within "translations" for each supported language:
+
+  * http://github.com/my_org/my_xblock/my_xblock/translations/ar/LC_MESSAGES/text.po
+  * http://github.com/my_org/my_xblock/my_xblock/translations/es/LC_MESSAGES/text.po
+  * http://github.com/my_org/my_xblock/my_xblock/translations/fr/LC_MESSAGES/text.po
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: ditto on the markup if this is an example file structure (are they files? or URLs?). Adding lead in sentences that explain what you're going to show them (as on line 51) is also a way to cue the reader. 
Note that these examples only provide language, and not locale, indicators. Is this intentional?

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 6]
```diff
@@ -20,7 +20,7 @@ LMS Requirements for XBlocks
 The LMS requires XBlocks to have the following properties.
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:24]: Please add a contents directive above line 16. This will create a bulleted list with linked cross references to the two headings in the file.
`.. contents::`
`:depth: 1`
`:local:`

Sorry, markup doesn't present this as it should be in the rst file. There should be three lines, one with the contents directive and its punctuation, one with the depth variable indented three spaces and then one with the local variable also indented three spaces...

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 7]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+* Compile your ".po" files into binary ".mo" files using the Gettext "msgfmt" command, and include these ".mo" files
+  alongside your ".po" files in your code project
+
+* Run the LMS and select your preferred language in order to observe the translated strings.
+
+* Note: In the absence of provided strings, the LMS will attempt to match any static strings marked for translation
+  within your XBlock using its own set of language locales and domains.  It is not recommended that you rely on the
+  LMS mechanism for I18N/L10N support, however -- there is no guarantee that your strings will be supported, and even
+  if a match is found, the translation may end up being incorrect given the context of your XBlock.
+
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:163]: Is the info on line 50 a prerequisite bit of preparation, and then the info on lines 56, 60, 66, 69, and 72 the steps in a procedure that users need to follow? If so, I'd suggest replacing the bullets (*) with items in an ordered list (#.) Each instruction should then start with a verb, like Create, Add, Compile, etc.
You could also add a subheading before line 56 to introduce the procedure, like "Add Translated Text Strings" or whatever makes sense to characterize the result of this set of activities.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 8]
```diff
@@ -40,4 +40,40 @@ The LMS requires XBlocks to have the following properties.
...
+  * http://github.com/my_org/my_xblock/my_xblock/translations
+
+* Create a set of subfolders (locales) and PO files (domains) within "translations" for each supported language:
+
+  * http://github.com/my_org/my_xblock/my_xblock/translations/ar/LC_MESSAGES/text.po
+  * http://github.com/my_org/my_xblock/my_xblock/translations/es/LC_MESSAGES/text.po
+  * http://github.com/my_org/my_xblock/my_xblock/translations/fr/LC_MESSAGES/text.po
+
+* Each domain file should include the set of strings to match for translations (msgid) and the set of corresponding
+  translations for the domain (msgstr)
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Add period at end of sentence (line 70 also).
I don't have the same background as most of the readers, but I think this could be clearer. Are msgid and msgstr the names of text files that are placed within the .po file? or somehow used to identify text within a single .po file?
Also, are "the set of strings to match for translations" the base language strings that the developer of the XBlock provided? (if it's an edX XBlock, that would be English, if it was from FUN, it might be French?), which are then paired up with the strings in the identified language?
Maybe something like, "Add the msgid and msgstr to each .po domain file. The msgid contains the set of original text strings for which you are provided translations, and the msgstr contains the corresponding set of translated text strings."
(I may be completely off base here, just trying to paraphrase and make simpler!)

**Comment by lamagnifica** [comment]
HI @mattdrayer this is great! As per my usual style, I pointed out everything I could think of. Use what you find helpful. I'll be happy to read it again.

**Comment by mattdrayer** [comment]
Great, thanks!  I'll fixup the PR and let you know when it's ready for another pass :)

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 6 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:24]: I added this as requested -- let me know if it needs to be repositioned.

**Comment by mattdrayer** [comment]
@lamagnifica @nedbat -- second draft pushed -- let me know what you think -- thanks!

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 9]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
   * ``video``
   * ``other``
 
+****************************
+Internationalization Support
+****************************
+
+The LMS is currently capable of supporting internationalization (i18n) and localization (l10n) of static XBlock chrome
+and control strings.  Translation of user-generated content stored as XBlock state is not currently supported.
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I wouldn't know what "chrome and control strings" are.  Maybe, "UI text included in your XBlock"?

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 10]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
...
+
+To present XBlock language translations in the LMS you must include the translated strings for your chosen "locale"
+in the GNU Gettext Portable Object file format.  Translated strings must be stored in a "domain" file named "text.po".
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:56]: "locale" and "domain" in quotes could use some explanation.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 11]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
...
+There is no limit on the number of locales that can be included with your XBlock.
+
+You can learn more about the GNU Gettext Portable Object file format using the following resources:
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I'd rather not send people off to these docs, they are very general, and are unlikely to help them with their task.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 12]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
...
+* https://www.drupal.org/node/1814954
+
+Each "text.po" domain file consists of one or more string/translation pairs for the language/locale.  Further, each
+translation pair consists of two fields: "msgid" for the base string, and "msgstr" for its corresponding translation.
+
+Adding Translated Strings to your XBlock
+****************************************
+
+#. Create a directory within your XBlock code project named "translations".  This directory should be located at the
+   same level in your code project as your XBlock implementation file.  For example:
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:86]: This makes me think we should update the XBlock SDK workbench-make-xblock script to create an empty translations directory, with a README linking to these docs.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 11 above)
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: They are going to need to install software to finish this work.  Is one of these links where they can get the tools?  I want to be sure that links are going someplace truly useful.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 13]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
...
+   "text.mo" files alongside your "text.po" files in your code project.
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/text.mo``
+   * ``../my_xblock/translations/ar/LC_MESSAGES/text.po``
+
+The resulting directory/file structure should look like this:
+
+.. code-block:: text
+
+    ../my_xblock
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Why does this start with ".." ?

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 14]
```diff
@@ -40,4 +44,83 @@ The LMS requires XBlocks to have the following properties.
...
+   * ``http://github.com/my_org/my_xblock/my_xblock/my_xblock.py``
+   * ``http://github.com/my_org/my_xblock/my_xblock/translations/``
+
+#. Create a set of locale-specific directories for each of your locales within this new "translations" directory.
+   Include an "LC_MESSAGES" directory with each locale directory.
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/``
+   * ``../my_xblock/translations/es-es/LC_MESSAGES/``
+
+#. Place each locale's "text.po" domain file within the corresponding "LC_MESSAGES" directory
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: We haven't said how to create the text.po files.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 12 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:86]: I have logged JIRA ticket WL-371 to take care of this -- we will pick it up in our next sprint.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 9 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Updated this statement to hopefully make more sense

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 10 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:56]: I've added definitions for these terms -- domain was actually a bit difficult to define based on all the docs and howtos I reviewed, so it might need to be changed a bit

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 11 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: The link to the GNU Gettext site provides an opportunity to download the software, yes.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 13 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Removed

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 14 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I added some information on how to do this -- I also found a nice (PHP) howto resource that reads similar to our doc and added that link as a reference.

**Comment by nedbat** [comment]
The gettext and wikipedia links are really bewildering.  For tools, can't we give people specific instructions about using i18n_tools for extraction and compilation?

**Comment by mattdrayer** [comment]
I've added some info about i18n-tools -- let me know what else is needed

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 15]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+Internationalization Support
+****************************
+
+The LMS is currently capable of supporting internationalization (i18n) and localization (l10n) of static UI text
+included in your XBlock -- also known as "chrome" or "labels".  Translation of user-generated content stored as XBlock
+state is not currently supported.
+
+To present XBlock language translations in the LMS you must include the translated strings for your chosen "locale"
+in the GNU Gettext Portable Object file format.  Translated strings must be stored in a "domain" file named "text.po".
+There is no limit on the number of locales that can be included with your XBlock, although not every locale may be
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Would it be correct to rephrase as, 
There is no limit... although your specific Open edX installation might not be configured to support every locale that you add" (?) (I'm a little confused by the word order in the second part of the sentence)

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 16]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+   * ``http://github.com/my_org/my_xblock/my_xblock/my_xblock.py``
+   * ``http://github.com/my_org/my_xblock/my_xblock/translations/``
+
+#. Create a set of locale-specific directories for each of your locales within this new "translations" directory.
+   Include an "LC_MESSAGES" directory with each locale directory.
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/``
+   * ``../my_xblock/translations/es-es/LC_MESSAGES/``
+
+#. Create a new domain file named "text.po".  You can use the Gettext ``xgettext`` command directly, or another tool
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: pet peeve: strike "new" (it's redundant after "Create")

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 17]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+   you support.  For more information on how to use these tools, see the following resources:
+   * Gettext: https://www.gnu.org/software/gettext/manual/gettext.html
+   * Gettext: http://phptal.org/manual/en/split/gettext.html
+   * Django: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#localization-how-to-create-language-files
+   * i18n-tools: https://github.com/edx/i18n-tools
+
+#. Open each "text.po" domain file and, for each "msgid" string, add a corresponding "msgstr" translation.  PO files
+   can be edited by hand, with a tool such as Pedit or Emacs, or through a third party service such as Transifex.
+
+#. Place each locale's "text.po" domain file within the corresponding "LC_MESSAGES" directory
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: add a period at the end of the sentence

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 18]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+        |       └── text.po
+        └── zh-cn
+            └── LC_MESSAGES
+                ├── text.mo
+                └── text.po
+
+You can now run the LMS and update your preferred language via Account Settings in order to observe the translated
+strings for your chosen locale.
+
+Note: In the absence of an available language locale and domain file, the LMS XBlock runtime will attempt to match
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: you can apply Sphinx note styling by using 
.. note:: 
in place of plain 
Note:

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 19]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+strings marked for translation within your XBlock using its own set of language locales and domains.  It is not
+recommended that you rely solely on the LMS mechanism for internationalization support, however -- there is no
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: replace  --  with a period

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 20]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+        |       └── text.po
+        ├── ru
+        |   └── LC_MESSAGES
+        |       ├── text.mo
+        |       └── text.po
+        └── zh-cn
+            └── LC_MESSAGES
+                ├── text.mo
+                └── text.po
+
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:154]: sweet!

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 21]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+   * Django: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#localization-how-to-create-language-files
+   * i18n-tools: https://github.com/edx/i18n-tools
+
+#. Open each "text.po" domain file and, for each "msgid" string, add a corresponding "msgstr" translation.  PO files
+   can be edited by hand, with a tool such as Pedit or Emacs, or through a third party service such as Transifex.
+
+#. Place each locale's "text.po" domain file within the corresponding "LC_MESSAGES" directory
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/text.po``
+   * ``../my_xblock/translations/es-es/LC_MESSAGES/text.po``
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:123]: I admit I did not read all the referenced material, but will readers understand that both conventions can be used, identifying both the language and the locale, or just the language? If you don't think that needs to be called out, that's fine, I just have always worked with files that had both, like your es-es example.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 22]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+   * ``http://github.com/my_org/my_xblock/my_xblock/translations/``
+
+#. Create a set of locale-specific directories for each of your locales within this new "translations" directory.
+   Include an "LC_MESSAGES" directory with each locale directory.
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/``
+   * ``../my_xblock/translations/es-es/LC_MESSAGES/``
+
+#. Create a new domain file named "text.po".  You can use the Gettext ``xgettext`` command directly, or another tool
+   of your choosing, such as Django's ``makemessages`` utility, or i18n-tools.  Repeat this process for each locale
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Move  "Repeat..." into its own step, so that in one step you create the file, and then in another step you iterate.

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 23]
```diff
@@ -40,4 +44,108 @@ The LMS requires XBlocks to have the following properties.
...
+   you support.  For more information on how to use these tools, see the following resources:
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Please end the sentence with a period rather than the colon (throughout). (Our preferred edX doc team style is to use complete sentences, ending with a period, to introduce lists, code samples, and so on. )

**Comment by lamagnifica** [comment]
a few more tweaks and suggestions. thanks so much for working on this! :+1: after you address what makes sense/you have time for...

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 12 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:86]: @nedbat -- see https://github.com/edx/xblock-sdk/pull/103

**Comment by mattdrayer** [comment]
@lamagnifica -- addressed your comments -- let me know if you'd like to see anything else -- would be great to merge this PR today

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 24]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+   * Gettext: https://www.gnu.org/software/gettext/manual/gettext.html
+   * Gettext: http://phptal.org/manual/en/split/gettext.html
+   * Django: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#localization-how-to-create-language-files
+   * i18n-tools: https://github.com/edx/i18n-tools
+
+#. Repeat the domain file creation process for each language/locale you support.
+
+   In the following example, we will use the i18n-tools utilites to generate a "text.po" file
+   #. Create an alternative configuration file containing the details for your particular XBlock project
+   #> Run ``i18n_tool extract``, which
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: change > to .

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 25]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+   of your choosing, such as Django's ``makemessages`` utility, or i18n-tools.  For more information on how to use
+   these tools, see the following resources.
+   * Gettext: https://www.gnu.org/software/gettext/manual/gettext.html
+   * Gettext: http://phptal.org/manual/en/split/gettext.html
+   * Django: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#localization-how-to-create-language-files
+   * i18n-tools: https://github.com/edx/i18n-tools
+
+#. Repeat the domain file creation process for each language/locale you support.
+
+   In the following example, we will use the i18n-tools utilites to generate a "text.po" file
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: add a period

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 26]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+        └── zh-cn
+            └── LC_MESSAGES
+                ├── text.mo
+                └── text.po
+
+You can now run the LMS and update your preferred language via Account Settings in order to observe the translated
+strings for your chosen locale.
+
+.. note:: In the absence of an available language locale and domain file, the LMS XBlock runtime will attempt to match
+strings marked for translation within your XBlock using its own set of language locales and domains.  However, it is
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: the lines after the first one need to be indented a couple of spaces

Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: I bumped them in two tab stops (4 spaces) to match the code block above

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 24 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Fixed

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 25 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst]: Done

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 6 above)
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:24]: looks fine!

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 27]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+****************************************
+
+#. Create a directory within your XBlock code project named "translations".  This directory should be located at the
+   same level in your code project as your XBlock implementation file.  For example:
+
+   * ``http://github.com/my_org/my_xblock/my_xblock/my_xblock.py``
+   * ``http://github.com/my_org/my_xblock/my_xblock/translations/``
+
+#. Create a set of language directories for each of your locales within this new "translations" directory.  You may
+   specify either a general language code or a specific language locale code for the name of each directory.  Include
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:92]: thanks!

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 28]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+   an "LC_MESSAGES" directory with each language directory.
+
+   * ``../my_xblock/translations/ar/LC_MESSAGES/``
+   * ``../my_xblock/translations/es-es/LC_MESSAGES/``
+
+#. Create a domain file named "text.po".  You can use the Gettext ``xgettext`` command directly, or another tool
+   of your choosing, such as Django's ``makemessages`` utility, or i18n-tools.  For more information on how to use
+   these tools, see the following resources.
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:100]: add a line feed after line 100 (the list won't render correctly if you don't separate it out)

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst [hunk 29]
```diff
@@ -40,4 +44,116 @@ The LMS requires XBlocks to have the following properties.
...
+   * Gettext: https://www.gnu.org/software/gettext/manual/gettext.html
+   * Gettext: http://phptal.org/manual/en/split/gettext.html
+   * Django: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#localization-how-to-create-language-files
+   * i18n-tools: https://github.com/edx/i18n-tools
+
+#. Repeat the domain file creation process for each language/locale you support.
+
+   In the following example, we will use the i18n-tools utilites to generate a "text.po" file.
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:109]: again, add a line feed before the list

**Comment by lamagnifica** [comment]
thanks Matt! :+1:

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 28 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:100]: ah, right -- fixed it

File: en_us/xblock-tutorial/source/edx_platform/edx_lms.rst (hunk 29 above)
Comment (mattdrayer) [inline en_us/xblock-tutorial/source/edx_platform/edx_lms.rst:109]: Fixed

**Comment by mattdrayer** [comment]
@nedbat -- GTG on this PR now?  Anything else you'd like to see?

**Comment by mattdrayer** [comment]
Caught up w/ @nedbat offline and we are good to move forward with this PR, so I am going ahead with the merge.
